source/connection-monitoring-and-pooling/tests/pool-clear-clears-waitqueue.yml

```diff
@@ -60,10 +59,9 @@ events:
   - type: ConnectionCheckOutFailed
     reason: connectionError
     address: 42
-  - type: ConnectionCheckedIn
-    address: 42
 ignore:
   - ConnectionPoolReady
   - ConnectionPoolCreated
   - ConnectionCreated
   - ConnectionReady
+  - ConnectionCheckedIn
```

Comment:
I think we also need to ignore ConnectionClosed event. Isn't the connection closed during the checkin process because we're checking in a "stale" connection?

Reply:
Yep, nice catch. This uncovered a bug in the Rust runner where we weren't asserting that extra events were seen after matching all the expected ones. Fixed.